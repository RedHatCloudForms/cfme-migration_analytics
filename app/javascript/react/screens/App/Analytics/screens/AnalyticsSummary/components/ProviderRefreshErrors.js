import React from 'react';
import PropTypes from 'prop-types';
import EmptyStateWithButton from '../../../components/EmptyStateWithButton';

const ProviderRefreshErrors = ({ providersWithRefreshErrors, onRetryClick }) => (
  <EmptyStateWithButton
    iconName="error-circle-o"
    title={__('Failed to refresh providers')}
    message={
      <React.Fragment>
        {__('Failed to refresh relationships and power states for the following providers:')}
        {providersWithRefreshErrors.map(provider => (
          <React.Fragment key={provider.id}>
            <br />
            <br />
            <strong>{provider.name}</strong>
            &nbsp;&nbsp;
            <a href={`/ems_infra/${provider.id}`}>{__('View provider')}</a>
            <br />
            {provider.last_refresh_error}
          </React.Fragment>
        ))}
      </React.Fragment>
    }
    buttonText={__('Try again')}
    onClick={onRetryClick}
  />
);

ProviderRefreshErrors.propTypes = {
  providersWithRefreshErrors: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.string,
      name: PropTypes.string,
      last_refresh_error: PropTypes.string
    })
  ).isRequired,
  onRetryClick: PropTypes.func.isRequired
};

export default ProviderRefreshErrors;
