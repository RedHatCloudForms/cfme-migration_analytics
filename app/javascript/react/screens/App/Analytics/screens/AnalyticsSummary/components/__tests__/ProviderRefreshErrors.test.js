import React from 'react';
import { shallow } from 'enzyme';
import ProviderRefreshErrors from '../ProviderRefreshErrors';

describe('provider refresh errors', () => {
  const getBaseProps = () => ({
    providersWithRefreshErrors: [
      { id: 1, name: 'Provider 1', last_refresh_error: 'some error here' },
      { id: 2, name: 'Provider 2', last_refresh_error: 'some error here' }
    ],
    onRetryClick: jest.fn()
  });

  test('renders correctly', () => {
    const component = shallow(<ProviderRefreshErrors {...getBaseProps()} />);
    expect(component).toMatchSnapshot();
  });
});
