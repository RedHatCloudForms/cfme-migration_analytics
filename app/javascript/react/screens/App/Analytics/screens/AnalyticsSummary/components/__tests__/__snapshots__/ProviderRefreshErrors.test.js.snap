// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`provider refresh errors renders correctly 1`] = `
<EmptyStateWithButton
  buttonText="Try again"
  href={null}
  iconName="error-circle-o"
  iconType="pf"
  message={
    <React.Fragment>
      Failed to refresh relationships and power states for the following providers:
      <React.Fragment>
        <br />
        <br />
        <strong>
          Provider 1
        </strong>
          
        <a
          href="/ems_infra/1"
        >
          View provider
        </a>
        <br />
        some error here
      </React.Fragment>
      <React.Fragment>
        <br />
        <br />
        <strong>
          Provider 2
        </strong>
          
        <a
          href="/ems_infra/2"
        >
          View provider
        </a>
        <br />
        some error here
      </React.Fragment>
    </React.Fragment>
  }
  onClick={[MockFunction]}
  title="Failed to refresh providers"
/>
`;
